### PR TITLE
[Term Entry] C++ vector .assign()

### LINKED ISSUE #4988: [Term Entry] C++ vector .assign()
### Reason/inspiration (optional)

"We would like a new entry on the `.assign()` term entry in Cpp under `Vector` concept. The entry should go in a new file under `content/cpp/concepts/vectors/terms/assign/assign.md.

The entry should include:

- An introduction to the concept
- A `Syntax` section that provides the syntax for the concept.
- An `Example` section that provides an example demonstrating the concept in use.

Please refer to the [concept entry template](https://github.com/Codecademy/docs/blob/main/documentation/concept-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue."

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by Great-Visions-Code**
### Description

- Added the vector.assign() term for C++

### Issue Solved

- Closes #4988 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

**Comment by Great-Visions-Code** [comment]
Ran scripts to have checks pass. Let me know if any other fixes are needed. Thank you.

**Review by cigar-galaxy82 — CHANGES_REQUESTED** [review]

**Comment by cigar-galaxy82** [comment]
Hi @Great-Visions-Code thank you for contributing to Codecademy docs, I have suggested a few changes please modify

File: content/cpp/concepts/vectors/terms/assign/assign.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    // Replacing current elements with elements from newNumbers
+    numbers.assign(newNumbers.begin(), newNumbers.end());
+    cout << "After assigning elements from newNumbers: ";
+    for (int num : numbers) {
+        cout << num << " ";
+    }
+    cout << endl;
+
+    return 0;
+}
```
[the remark this answers is not in the document]
Comment (Great-Visions-Code) [reply content/cpp/concepts/vectors/terms/assign/assign.md:63]: @cigar-galaxy82 quick question. There is an added line on line 52 after the for loop. Should I also make that change to line 61? Add a line after line 60, after the 2nd for loop so they match?

**Comment by Great-Visions-Code** [comment]
Hi @cigar-galaxy82. I have made the changes suggested. Let me know how it looks on your end and if any other modifications are needed. Thank you.

File: content/cpp/concepts/vectors/terms/assign/assign.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (cigar-galaxy82) [reply content/cpp/concepts/vectors/terms/assign/assign.md]: I'll recommend adding a gap, I have made that changes

**Review by cigar-galaxy82 — APPROVED** [review]

**Review by mamtawardhani — APPROVED** [review]
Thank you for your contribution to Codecademy Docs @Great-Visions-Code 😄 

The entry is ready to be merged! 🚀

**Pull request merged**